opensearch/src/test/java/org/opensearch/sql/opensearch/data/type/OpenSearchDataTypeTest.java

```diff
@@ -213,26 +218,28 @@ public void fields_and_properties_are_readonly() {
   @Test
   // Test and type added for coverage only
   public void of_null_MappingType() {
-    assertThrows(IllegalArgumentException.class, () -> OpenSearchDataType.of(MappingType.Invalid));
+    assertNotNull(OpenSearchDataType.of(MappingType.Invalid));
```

Comment:
Why?

Reply:
I believe I had to change this due to some of the changes for the datatype refactor. It was not throwing an exception anymore. However, I didn't think that this test mattered since the comment above it mentions that it was only added for code coverage.

Reply:
I think we should throw an error like the test has.  
It's a better API to reject rather than silently return an invalid/unsupported object (null) that might fail unexpectedly later on.